### PR TITLE
added test script for python

### LINKED ISSUE #74: Python test script
Have a look at https://github.com/sevehub/trurl, JSON data (test.json) and the script (test.py) are separated. It also shows the return codes.

Let me know if you prefer this solution or if you are happier with the existing Perl scripts.

### PR DESCRIPTION
**Pull request by havbon**
I have now taken all the tests, both json and not and put them in tests.json
Then i have disabled stderr testing for now, they are still present in json files for future.

I added the following features:

- run just a specific test case
- verify the exit code (to better test errors)

In order to run specific cases simply run test.py index,index,index (comma sep no spaces)
I planted a few issues for testing purposes.

closes #74

### CODE REVIEW

**Comment by havbon** [comment]
I was going to say that i havent fully integrated the script into CI and removed the Perl script because i ran out of time, but then sonatype freaked out. I guess i will check it out tomorrow

File: tests.json [hunk 1]
```diff
@@ -0,0 +1,610 @@
+[
+    {
+        "stdin": "example.com",
```
Comment (bagder) [inline tests.json:3]: This is not really `stdin`, I think maybe `cmdline` would be more appropriate? I would imagine that we actually want to eventually be able to provide data on stdin for tests.

File: test.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        self.cmdline = testCase["cmdline"]
+        self.expected = stripDict(testCase["expected"])
+        self.commandOutput: CommandOutput = None
+        self.testPassed: bool = False
+
+    def runCommand(self, stdinkeyword: str):
+        # returns false if the command does not contain keyword
+        if self.cmdline.find(stdinkeyword) == -1:
+            return False
+        output = subprocess.run(
```
[the remark this answers is not in the document]
Comment (havbon) [reply test.py:52]: @sonatype-lift ignore

**Comment by havbon** [comment]
@sonatype-lift ignoreall

**Comment by havbon** [comment]
after lots of tweaks and adjustments I have replaced the entire perl test suite to json, and switched the script to python. Checking stderr is somewhat implemented. I decided to replace the filter system, if `python.exe test.py --get` is run then it will only test the commands with `--get` in the arguments, this allows more flexible and faster testing

**Comment by bagder** [comment]
I don't see the `--json` tests converted. Am I just blind?

**Comment by havbon** [comment]
> I don't see the `--json` tests converted. Am I just blind?

Everything should be fixed now, I hope

**Comment by bagder** [comment]
> Conflicting files test.pl 

Does this mean that there are a few recently added tests that are not converted over? I understand it is a little annoying but we have been adding tests to test.pl while you've worked on the python version and I'm just keen on making sure that we don't miss out on those. It will probably help if you don't remove the perl version in your PR so that we can make sure they both work equally fine before we remove the perl version in a separate PR.

**Comment by havbon** [comment]
Okay i squashed and pushed, it should be fine now. I manually went through each test and I am quite certain I got every one, its also the same number of tests in perl and the json file.

**Review by bagder — APPROVED** [review]

**Comment by bagder** [comment]
Thanks!

**Pull request merged**